packages/hardhat-core/src/internal/hardhat-network/provider/vm/rethnet.ts

```diff
@@ -520,3 +558,12 @@ export class RethnetAdapter implements VMAdapter {
     return undefined;
   }
 }
+
+// For compatibility with solidity-coverage that attaches a listener to the step event.
```

Comment:
I wonder whether the VMAdapter interface can extend this interface to avoid duplication? That way we can add all of the logic for supporting solidity coverage in one place and call that from the rethnet and ethereumjs implementations

Reply:
I also thought something similar: abstracting this at the context level somehow (e.g., a `context.onStep`) function. That also has the advantage of making the need for this functionality more explicit.

But on the other hand, I like that the current approach doesn't need any changes to the ethereumjs adapter, so I slightly prefer to just do it this way.